src/app.tsx

```diff
@@ -0,0 +1,438 @@
+              <div className="relative p-6 flex-auto">
+                <p className="my-4 text-blueGray-500 text-lg leading-relaxed">
+                  {modalContent}
+                </p>
+              </div>
+              {/*footer*/}
+              <div className="flex items-center justify-end p-6 border-t border-solid border-blueGray-200 rounded-b">
+                <button
+                  className="text-red-500 background-transparent font-bold uppercase px-6 py-2 text-sm outline-none focus:outline-none mr-1 mb-1 ease-linear transition-all duration-150"
+                  type="button"
+                  onClick={props.onClick}
+                >
+                  Close
+                </button>
+              </div>
+            </div>
+          </div>
+        </div>
+        <div className="opacity-25 fixed inset-0 z-40 bg-black"></div>
+      </>
+    );
+  } else {
+    return null;
+  }
+};
+
+// bug component
+interface BugProps {
+  key: number;
+  id: number;
+  bugType: string;
+  onClick?: () => void;
+}
+
+const Bug: React.FC<BugProps> = (props) => {
+  const [x, setX] = useState(VIEW_WIDTH * Math.random());
+  const [y, setY] = useState(VIEW_HEIGHT * Math.random());
+  const [targetX, setTargetX] = useState(VIEW_WIDTH * Math.random());
+  const [targetY, setTargetY] = useState(VIEW_HEIGHT * Math.random());
+  const [isOnHover, setIsOnHover] = useState(false);
+
+  // useRef hook enables values to persist in successive renders
+  const posIntervalID = useRef(0);
+  const moveIntervalID = useRef(0);
+
+  useEffect(() => {
+    // when component mounts
+    const newPos = () => {
+      setTargetX(VIEW_WIDTH * Math.random());
+      setTargetY(VIEW_HEIGHT * Math.random());
+    };
+
+    // component finds a new target position every 2 second
+    posIntervalID.current = window.setInterval(newPos, 2000);
+
+    return () => {
+      // when component unmounts
+      clearInterval(posIntervalID.current);
+    };
+  }, []);
+
+  useEffect(() => {
+    const move = () => {
+      const moveRate = 0.01;
+
+      setX((x) => (moveRate * targetX + x) / (1 + moveRate));
+      setY((y) => (moveRate * targetY + y) / (1 + moveRate));
+    };
+
+    // component changes x, y values every 20ms
+    moveIntervalID.current = window.setInterval(move, 20);
+
+    return () => {
+      clearInterval(moveIntervalID.current);
+    };
+  }, [targetX, targetY]);
+
+  const changeHoverStyle = () => {
+    setIsOnHover((prevState) => !prevState);
+  };
+
+  const bugImgStyle = {
+    filter: "none",
+  };
+
+  const hoverBugImgStyle = {
+    cursor: "pointer",
+    filter: "drop-shadow(2px 2px 4px #cc0000)",
+  };
+
+  return (
+    // foreign object is used to wrap the image and render it on svg canvas
+    <foreignObject x={x} y={y} width="120" height="120" onClick={props.onClick}>
+      <img
+        src={BUGS[props.bugType].imgSrc}
+        alt={`${props.bugType}`}
+        style={isOnHover ? hoverBugImgStyle : bugImgStyle}
```

Comment:
Use CSS for hover styling. It is more efficient.

https://developer.mozilla.org/en-US/docs/Web/CSS/:hover

https://tailwindcss.com/docs/hover-focus-and-other-states#hover